src/zenml/materializers/numpy_materializer.py

```diff
@@ -145,4 +144,53 @@ def extract_metadata(
             "min": min_val,
             "max": max_val,
         }
-        return {**base_metadata, **numpy_metadata}
+        return numpy_metadata
+
+    def extract_text_metadata(
+        self, arr: "NDArray[Any]"
+    ) -> Dict[str, "MetadataType"]:
+        """Extracts text metadata from a numpy array.
+
+        Args:
+            arr: The numpy array to extract metadata from.
+
+        Returns:
+            A dictionary of metadata.
+        """
+        text = " ".join(arr)
+        words = text.split()
```

Comment:
This I don't get, you join the whole array into a giant string but then immediately split it again, shouldn't there be an easier way to do that such as `arr.tolist()`? Or is it because each array element could have multiple words in it? (I'm not familiar with text numpy arrays at all)

Reply:
because in first join its series of phrases, so we join every phrase to give us one big text that we then split based on words? would it make a different if we iterate and split every phrase?

Reply:
Nah it's fine then I guess, no point in performance optimization for something like this